packages/core/src/components/Dropdown/Dropdown.types.ts

```diff
@@ -150,7 +150,7 @@ export interface DropdownComponentProps extends CustomMenuBaseProps, CustomOptio
   /**
    * custom menu render function
    */
-  menuRenderer?: React.ReactElement;
+  menuRenderer?: React.ReactElement | ((props: CustomMenuProps) => React.ReactElement);
```

Comment:
I believe the type here is a bit more complex
can you check menu.jsx file?
its `rendererProps`.
everything which is passed to `{Renderer && Renderer(rendererProps)}`
so apparently we're also passing by mistake `innerProps`, the `onScroll`, etc.

what a mess 🤦🏼‍♂️
you can give it an any in the meantime if stuff gets too complicated (see what I did in `valueRenderer`, they needed the `DropdownOption` stuff, so I had to add it, but basically `DropdownOption` is `any`, so the all `props` became `any` at the bundle, so I've worked hard for nothing 🙃)

Reply:
I agree that it is a mess, but itsn't it still better to keep it this way then changing it to `any`?

Reply:
they would have TS issues in the current way if they try to consume something which isn't in your type
we can try that and if they complain we'll release a fix

Reply:
I tested it with current implementations and it worked properly, I will keep an eye on it and change if necessary